src/models/thread.ts

```diff
@@ -197,6 +198,10 @@ export class Thread extends ReadReceipt<ThreadEmittedEvents, ThreadEventHandlerM
             this._currentUserParticipated = false;
             this.emit(ThreadEvent.Delete, this);
         } else {
+            if (this.lastEvent?.getId() === event.getAssociatedId()) {
+                // XXX: If our last event got redacted we query the server for the last event once again
+                this.processRootEventPromise = undefined;
```

Comment:
Are there any adverse effects that could occur if the new root event promise, racing with the previous one, completes before the previous one?

Reply:
Good point, I doubt it, but awaiting before resetting wouldn't hurt